### PR TITLE
Refactor master to use JsonUtility and KeyProvider

### PR DESCRIPTION
**Pull request by tttzach**
Replace all raw JSON writing with `JsonUtility.sendJson` and hide Client ID in AuthenticationVerifier.

### CODE REVIEW

File: src/test/java/ClientIDServletTest.java [hunk 1]
```diff
@@ -22,11 +21,12 @@
 /** Test Client ID Servlet to ensure response contains a defined OAuth 2.0 client ID */
 @RunWith(JUnit4.class)
 public final class ClientIDServletTest extends ServletTestBase {
-  private static final ClientIDServlet servlet = new ClientIDServlet();
+
+  private static final ClientIDServlet servlet = new ClientIDServlet("sampleValue");
```
Comment (austinteshuba) [inline src/test/java/ClientIDServletTest.java:25]: Declare a constant for `"sampleValue"`

Comment (tttzach) [reply src/test/java/ClientIDServletTest.java:25]: Good call, thanks!

File: src/main/java/com/google/sps/model/AuthenticationVerifier.java [hunk 2]
```diff
@@ -14,15 +14,20 @@
...
-  // OAuth 2.0 Client ID
-  // TODO: Hide Client ID (Issue #72)
-  String CLIENT_ID = "12440562259-mf97tunvqs179cu1bu7s6pg749gdpked.apps.googleusercontent.com";
-
+  /**
+   * Retrieves hidden Client ID from Secret Manager
+   *
+   * @return String representing Client ID value defined in Secret Manager
+   */
+  public static String getClientId() throws IOException {
```
[the remark this answers is not in the document]
Comment (tttzach) [reply src/main/java/com/google/sps/model/AuthenticationVerifier.java:28]: Thanks for letting me know!

[the remark this answers is not in the document]
Comment (tttzach) [reply src/main/java/com/google/sps/model/AuthenticationVerifier.java]: `RESOLVED`

File: src/test/java/ClientIDServletTest.java (hunk 1 above)
Comment (tttzach) [reply src/test/java/ClientIDServletTest.java]: `RESOLVED`

File: src/main/java/com/google/sps/model/AuthenticationVerifier.java [hunk 3]
```diff
@@ -14,15 +14,20 @@
...
 
+import com.google.sps.utility.KeyProvider;
 import java.io.IOException;
 import java.security.GeneralSecurityException;
 
 /** Contract for verifying user with Google Sign in API */
 public interface AuthenticationVerifier {
-  // OAuth 2.0 Client ID
-  // TODO: Hide Client ID (Issue #72)
-  String CLIENT_ID = "12440562259-mf97tunvqs179cu1bu7s6pg749gdpked.apps.googleusercontent.com";
```
Comment (gbuenoandrade) [inline src/main/java/com/google/sps/model/AuthenticationVerifier.java:24]: If it's indeed important to hide this value, we might to generate a new one as this value will be public "forever"

File: src/main/java/com/google/sps/utility/KeyProvider.java [hunk 4]
```diff
@@ -29,7 +29,11 @@
 
   /** Constructor to create instance with file linked to default path. */
   public KeyProvider() {
-    this.file = new File("src/main/resources/KEYS.json");
+    if (System.getProperty("user.dir").contains("target")) {
```
Comment (gbuenoandrade) [inline src/main/java/com/google/sps/utility/KeyProvider.java:32]: Please add a comment explaining why this is being necessary

File: src/main/java/com/google/sps/model/AuthenticationVerifier.java (hunk 3 above)
Comment (tttzach) [reply src/main/java/com/google/sps/model/AuthenticationVerifier.java:24]: Agreed, I will do the same for the API key as well.

File: src/main/java/com/google/sps/utility/KeyProvider.java (hunk 4 above)
Comment (tttzach) [reply src/main/java/com/google/sps/utility/KeyProvider.java:34]: I have added a comment, let me know if that makes sense!

File: src/main/java/com/google/sps/model/AuthenticationVerifier.java (hunk 3 above)
Comment (tttzach) [reply src/main/java/com/google/sps/model/AuthenticationVerifier.java:24]: `RESOLVED`

File: src/main/java/com/google/sps/utility/KeyProvider.java (hunk 4 above)
Comment (gbuenoandrade) [reply src/main/java/com/google/sps/utility/KeyProvider.java:34]: Tks!
RESOLVED

**Pull request merged**